### ISSUE TITLE
[Problem]: Sound only works from one docker instance at a time

### ISSUE THREAD

**Issue by nhangeland** [mikebrady/shairport-sync#1694]
### What happened?

Trying to run SPS with AP2 in docker using the same sound card with multiple outputs (Scarlett 18i20).
Audio works fine for the first instance of SPS I play to, but the subsequent instances I get no audio. They work just fine individually.

### Configuration Information.

asound.conf:
```
pcm_slave.outs {
        pcm "hw:USB"
        rate 44100
        channels 10
}

pcm.livingroom {
        type plug
        slave.pcm {
                type dshare
                ipc_key 4242
                slave outs
                bindings.0 0
                bindings.1 1
        }
}

pcm.office {
        type plug
        slave.pcm {
                type dshare
                ipc_key 4242
                slave outs
                bindings.0 2
                bindings.1 3
        }
}

pcm.kitchen {
        type plug
        slave.pcm {
                type dshare
                ipc_key 4242
                slave outs
                bindings.0 4
        }
}
.....
```
docker-compose:
```
services:
  office:
    image: mikebrady/shairport-sync:development
    restart: always
    devices:
      - "/dev/snd"
    volumes:
      - ./config/office.conf:/etc/shairport-sync.conf
      - ./config/asound.conf:/etc/asound.conf
    networks:
      spsnet:
        ipv4_address: 10.0.0.202
  
  kitchen:
    image: mikebrady/shairport-sync:development
    restart: always
    devices:
      - "/dev/snd"
    volumes:
      - ./config/kitchen.conf:/etc/shairport-sync.conf
      - ./config/asound.conf:/etc/asound.conf
    networks:
      spsnet:
        ipv4_address: 10.0.0.203
  living_room:
    image: mikebrady/shairport-sync:development
    restart: always
    devices:
      - "/dev/snd"
    volumes:
      - ./config/living-room.conf:/etc/shairport-sync.conf
      - ./config/asound.conf:/etc/asound.conf
    networks:
      spsnet:
        ipv4_address: 10.0.0.206
     
networks:
  spsnet:
    name: spsnet
    driver: macvlan
    driver_opts:
      parent: ens18
    ipam:
      config:
        - subnet: 10.0.0.0/24+
          gateway: 10.0.0.133
```
sps config:
```
general =
{
	name = "Kitchen";
	// Im using a 10 port offest for each instance and the same for the id offset
	port = 7040;
	airplay_device_id_offset = 40;
};

alsa =
{
	output_device = "kitchen";
};
```

### How did you install Shairport Sync?

Docker

### Check previous issues

- [X] Confirm

**Comment by nhangeland** [comment]
So the issue seems to be my alsa config or something in that direction. I get the following error using this setup in the containers not playing sound: ```(snd_pcm_dshare_open) unable to open slave```
Shamelessly tagging @noelhibbard as I see that he's quite an expert at these things.

**Comment by mikebrady** [comment]
> Shamelessly tagging @noelhibbard as I see that he's quite an expert at these things.

For sure!

**Comment by noelhibbard** [comment]
I've had zero luck getting ALSA to work in docker unless they are discrete cards. If I bust up the channels of a single card like you have I get the same issue you are (one at a time works). My workaround is to use PulseAudio to bust up the channels up.

Check out my repo here:
https://github.com/noelhibbard/shairport-sync-docker

I really wish I could get it to work with pure ALSA though.

**Comment by nhangeland** [comment]
"No such entity" was caused by a mismatch in the configured sink name in system.pa and sps config. I now got it working! Awesome if you could update your example repo with the alsa config example and the missing env var.

**Comment by noelhibbard** [comment]
> "No such entity" was caused by a mismatch in the configured sink name in system.pa and sps config. I now got it working! Awesome if you could update your example repo with the alsa config example and the missing env var.

I used to use the env var but at some point it stopped working in my setup and the only way to get it working again was to specify the PA server in my SPS config. I also ditched cookies and set the PA server to accept anon connections just to simplify permissions problems. I'll add the env var back even though it's redundant and I'll add a little ALSA sample.

Glad you got it working. Still wish we could go pure ALSA though.

**Comment by nhangeland** [comment]
Went with the anon option so I removed the env var for this. Your shairport-sync-instance1/2.conf does not specify PA as audio backend, and the pa section seems to be missing pa server, so this is the reason for my confusion.

**Comment by noelhibbard** [comment]
> Went with the anon option so I removed the env var for this. Your shairport-sync-instance1/2.conf does not specify PA as audio backend, and the pa section seems to be missing pa server, so this is the reason for my confusion.

That is strange, I could have sworn I updated all of that. Yeah that is really confusing. I'll get that fixed right now. Thanks for pointing that out.

**Comment by noelhibbard** [comment]
Okay I am refreshing my memory as I go through my current config. When I finally got around to using the official docker image with built in PA support my environment stopped working. I noticed that sps was trying to use UNIX:/var/run/pulse/native for the server regardless of what I set in the PULSE_SERVER env var. To work around that I changed my server.pa to put the socket in /var/run/pulse/native and then removed the env var and didn't bother specifying the server in the sps config. The line in my system.pa looks like this:
```
load-module module-native-protocol-unix auth-anonymous=1 socket=/var/run/pulse/native
```

So, there are several ways to do this, I'd like to make my documentation as dummy proof as possible, not just for others but also for myself. Hahaha. I guess I could be really redundant and set the socket in the PA config and then set the server in the env and in the sps config. Hmmm
